FE/src/script.js

```diff
@@ -189,51 +194,59 @@ searchBar.addEventListener("input", () => {
   });
 });
 
-let draggingTask;
-
-stateList.addEventListener("dragstart", (event) => {
-  draggingTask = event.target;
-  event.dataTransfer.setData("text/plain", null);
+stateList.addEventListener("dragstart", (e) => {
+  draggingTask = e.target;
+  e.dataTransfer.setData("text/plain", null);
+  draggingStart = e.clientX;
 });
 
-stateList.addEventListener("dragleave", (event) => {
-  const dropTarget = getDropTarget(event.target, event.clientY);
-  if (dropTarget) {
-    dropTarget.classList.remove("drag-over");
-  }
-});
 
-stateList.addEventListener("dragover", (event) => {
-  event.preventDefault();
-  const dropTarget = getDropTarget(event.target, event.clientY);
+
+stateList.addEventListener("dragover", (e) => {
+  e.preventDefault();
+  const dropTarget = getDropTarget(e.target, e.clientX);
   if (dropTarget) {
     dropTarget.classList.add("drag-over");
   }
 });
 
-stateList.addEventListener("drop", (event) => {
-  event.preventDefault();
-  const dropTarget = getDropTarget(event.target, event.clientY);
-  if (dropTarget) {
-    const draggingIndex = Array.from(stateList.children).indexOf(draggingTask);
-    const dropIndex = Array.from(stateList.children).indexOf(dropTarget);
-    let tempTask = storageList[draggingIndex];
-    if (draggingIndex < dropIndex) {
-      stateList.insertBefore(draggingTask, dropTarget.nextSibling);
-    } else {
-      stateList.insertBefore(draggingTask, dropTarget);
-    }
+stateList.addEventListener("drop", (e) => {
+  e.preventDefault();
+  const dropTarget = getDropTarget(e.target, e.clientY);
+  //if (dropTarget) {
+  const draggingIndex = Array.from(stateList.children).indexOf(draggingTask);
+  const dropIndex = Array.from(stateList.children).indexOf(dropTarget);
+  if (draggingStart - 120 > e.clientX) {
+    // dragging to the left
+    draggingTask.remove();
     storageList.splice(draggingIndex, 1);
-    storageList.splice(dropIndex, 0, tempTask);
     localStorage.setItem("list", JSON.stringify(storageList));
+  } else if (draggingStart + 120 < e.clientX) {
+    // dragging to the right
+    draggingTask.style.textDecoration === "line-through"
+      ? (draggingTask.style.textDecoration = "none")
+      : (draggingTask.style.textDecoration = "line-through");
   }
+  //}
   draggingTask.classList.remove("dragging");
   document.querySelectorAll(".drag-over").forEach((dropTarget) => {
     dropTarget.classList.remove("drag-over");
   });
 });
 
-function getDropTarget(target, y) {
+function getDropTarget(parent, y) {
+  for (const task of Array.from(parent.children)) {
+    if (task === draggingTask) continue;
```

Comment:
```suggestion
    if (task !== draggingTask)
     return;
```
Faz me mais sentido esta lógica assim